Update multi-project docs for v1.5.0-rc1 (!) (#3171)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-properties/constraints.md

datatype: "{dictionary}"
---

:::info New functionality
This functionality is new in v1.5.
:::

> Constraints are a feature of many data platforms. When specified, the platform will perform additional validation on data as it is being populated in a new table, or inserted into a preexisting table. If the validation fails, the table creation or update fails, the operation is rolled back, and you will see a clear error message.

Reviewer: ```suggestion
Constraints are a feature of many data platforms. When specified, the platform will perform additional validation on data as it is being populated in a new table or inserted into a preexisting table. If the validation fails, the table creation or update fails, the operation is rolled back, and you will see a clear error message.
```

When enforced, a constraint guarantees that you will never see invalid data in the table materialized by your model. Enforcement varies significantly by data platform.

Constraints require the declaration and enforcement of a model [contract](resource-configs/contract)

> **Constraints are never applied on models materialized as `view`** (or `ephemeral`). Only `table` and `incremental` models support the application and enforcement of constraints.

Reviewer: ```suggestion
**Constraints are never applied on `ephemeral` models or those materialized as `view`**. Only `table` and `incremental` models support applying and enforcing constraints.
```
